Library/Homebrew/extend/os/linux/linkage_checker.rb

```diff
@@ -26,9 +26,32 @@ class LinkageChecker
     libstdc++.so.6
   ].freeze
 
+  def display_deprecated_warning
+    return unless @libcrypt_found
+
+    opoo "Linkage to libcrypt.so.1 is deprecated and will fail linkage checks in Homebrew 3.??.0.\n" \
+         "Migrate to libcrypt.so.2 in the libxcrypt formula instead."
```

Comment:
```suggestion
    opoo "Linkage to libcrypt.so.1 is deprecated! \n" \
         "Migrate to libcrypt.so.2 in the libxcrypt formula instead."
```

or ideally just:

```suggestion
    odeprecated "linkage to libcrypt.so.1", "link to libcrypt.so.2 from the libxcrypt formula"
```

Reply:
I intentionally do not want `odeprecated`'s raise-on-`HOMEBREW_DEVELOPER` behaviour at this time. I'd like the warning to appear in our CI but not actually fail linkage yet.

Reply:
>  Or maybe actually have a stepping stone of some opt-in raise-to-error

Modified it now with that.

Reply:
Could this use `odeprecated` when `--strict` is passed? Feels a bit weird to have a deprecation that's not using it. Alternatively, a flag for `odeprecated` to make it require more than `HOMEBREW_DEVELOPER` in this case?

Reply:
Adjusted this now. The English is a bit weird since it says "Calling linkage", but other than that it works.